### PR TITLE
[18.0][FIX] web_pivot_computed_measure: fix tests

### PR DESCRIPTION
**Pull request by kobros-tech**
This fix is the cause for not merging a new migration for web_responsive module to version 18.0.
The migration PR is #3037

### CODE REVIEW

**Comment by kobros-tech** [comment]
I am finding this error all the way:

2024-12-25 07:26:39,542 54005 INFO web_responsive_9 odoo.addons.web_pivot_computed_measure.tests.test_ui_pivot.TestUIPivot.test_ui: waiting for threads: [] 
2024-12-25 07:26:39,544 54005 INFO web_responsive_9 odoo.addons.web_pivot_computed_measure.tests.test_ui_pivot: ====================================================================== 
2024-12-25 07:26:39,544 54005 ERROR web_responsive_9 odoo.addons.web_pivot_computed_measure.tests.test_ui_pivot: FAIL: TestUIPivot.test_ui
Traceback (most recent call last):
  File "/home/kobros/Workspace/odoo18/web/web_pivot_computed_measure/tests/test_ui_pivot.py", line 62, in test_ui
    self.start_tour(
  File "/home/kobros/Workspace/odoo18/odoo/odoo/tests/common.py", line 2076, in start_tour
    return self.browser_js(url_path=url_path, code=code, ready=ready, timeout=timeout, success_signal="tour succeeded", **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kobros/Workspace/odoo18/odoo/odoo/tests/common.py", line 2052, in browser_js
    self.fail('%s\n\n%s' % (message, error))
AssertionError: The test code "odoo.startTour('web_pivot_computed_measure_tour', {"stepDelay": 100, "keepWatchBrowser": false, "debug": false, "startUrl": "/odoo", "delayToCheckUndeterminisms": 0})" failed

UncaughtTypeError: Cannot read properties of undefined (reading '__computed_id')
    at PivotModel.addComputedMeasure (http://127.0.0.1:8069/web/assets/06ae128/web.assets_backend_lazy.min.js:375:229)
    at DropdownItemCustomMeasure.addMeasure (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:14672:150)
    at Object.mainEventHandler (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:1603:77)
    at HTMLButtonElement.listener (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:762:15)
    at dispatch (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15204:71)
    at triggerClick (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15132:94)
    at _pointerUp (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15187:24)
    at async _click (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15141:456)
    at async Object.click (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15197:187)
    at async TourHelpers.click (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:14733:72)
	at addComputedMeasure (http://127.0.0.1:8069/web/assets/06ae128/web.assets_backend_lazy.min.js:374:228)
	at addMeasure (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:14671:149)
	at mainEventHandler (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:1602:76)
	at listener (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:761:14)
	at dispatch (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15203:70)
	at triggerClick (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15131:93)
	at _pointerUp (http://127.0.0.1:8069/web/assets/aeba509/web.assets_web.min.js:15186:23)

This external module was important for the test:
websocket-client

with it being installed test failed, without it test didn't run!

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 1]
```diff
@@ -216,20 +218,31 @@ patch(PivotModel.prototype, {
...
-                    });
-                    toAnalyze.push(cm.field1, cm.field2);
-                    const toEnableFields = [];
-                    if (!this.metaData.fields[cm.field1].__computed_id) {
-                        toEnableFields.push(cm.field1);
-                    }
-                    if (!this.metaData.fields[cm.field2].__computed_id) {
-                        toEnableFields.push(cm.field2);
+                if (fieldDef) {
+                    if (fieldDef.__computed_id) {
```
Comment (chienandalu) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: Isn't it the same? You can use the shorter form. Anyway this change isn't needed IMO

```js
if (fieldDef?.__computed_id)

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 2]
```diff
@@ -275,17 +288,27 @@ patch(PivotModel.prototype, {
         const fieldNames = Object.keys(this.metaData.fields);
         for (const fieldName of fieldNames) {
             const field = this.metaData.fields[fieldName];
-            if (field.__computed_id) {
```
Comment (chienandalu) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js:278]: Minimize the diff
```suggestion
            if (field?.__computed_id) {

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 3]
```diff
@@ -275,17 +288,27 @@ patch(PivotModel.prototype, {
...
+                            );
+                    }
+                } else {
+                    console.warn(
+                        `Field "${fieldName}" is missing __computed_id`,
+                        field
                     );
                 }
+            } else {
+                console.error(`Field "${fieldName}" is undefined in metaData.fields`);
```
Comment (chienandalu) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: Are these warnings and errors of any use? If this is a corner case where the code is executed but those variables aren't ready, probably the best thing to do is just to ignore it...

File: web_pivot_computed_measure/tests/res_users_fake.py [hunk 4]
```diff
@@ -1,12 +1,7 @@
-# Copyright 2022 Tecnativa - Carlos Roca
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
-
-
```
Comment (chienandalu) [inline web_pivot_computed_measure/tests/res_users_fake.py:4]: :thinking:

File: web_pivot_computed_measure/tests/test_ui_pivot.py [hunk 5]
```diff
@@ -10,18 +11,48 @@ class TestUIPivot(common.HttpCase):
...
+        #     {
+        #         "name": "x_user_year_now",
+        #         "model": "res.users",
+        #         "field_description": "Year Now",
+        #         "ttype": "integer",
+        #         "store": True,
+        #         "model_id": res_users_model_id,
+        #     }
+        # )
+
```
Comment (chienandalu) [inline web_pivot_computed_measure/tests/test_ui_pivot.py]: Just remove it?
```suggestion
```

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js (hunk 1 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: yes, sure

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js (hunk 3 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: I think the test is not aligned with the code, there maybe some promises not functioning.

I tried but I failed as I am not the maintainer, I need help to improve the test case.

File: web_pivot_computed_measure/tests/res_users_fake.py (hunk 4 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/tests/res_users_fake.py:4]: sorry sorry, I don't mean to ignore the author :)

File: web_pivot_computed_measure/tests/test_ui_pivot.py (hunk 5 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/tests/test_ui_pivot.py]: @chienandalu 

I am leaving my comment, as the original code is using a model and we get asked about the fields for this model.

We must create fake fields as while migrating another module errors came from the tests here.

user_year_now, and user_year_born fields

Check this PR please: #3037

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js (hunk 3 above)
Comment (chienandalu) [reply web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: @CarlosRoca13 is on holidays these days. Maybe when he comes back could give you some hints :wink:

File: web_pivot_computed_measure/tests/test_ui_pivot.py (hunk 5 above)
Comment (chienandalu) [reply web_pivot_computed_measure/tests/test_ui_pivot.py]: yeah, kinda makes sense without having an in depth look

**Comment by kobros-tech** [comment]
@chienandalu see here:

    at async TourHelpers.click (http://127.0.0.1:8069/web/assets/1a839a2/web.assets_web.min.js:10674:72) 
2024-12-30 15:24:55,997 250 ERROR odoo odoo.addons.web_pivot_computed_measure.tests.test_ui_pivot: FAIL: TestUIPivot.test_ui
Traceback (most recent call last):
  File "/__w/web/web/web_pivot_computed_measure/tests/test_ui_pivot.py", line 63, in test_ui
    self.start_tour(
  File "/opt/odoo/odoo/tests/common.py", line 2076, in start_tour
    return self.browser_js(url_path=url_path, code=code, ready=ready, timeout=timeout, success_signal="tour succeeded", **kwargs)
  File "/opt/odoo/odoo/tests/common.py", line 2052, in browser_js
    self.fail('%s\n\n%s' % (message, error))
AssertionError: The test code "odoo.startTour('web_pivot_computed_measure_tour', {"stepDelay": 100, "keepWatchBrowser": false, "debug": false, "startUrl": "/odoo", "delayToCheckUndeterminisms": 0})" failed

UncaughtTypeError: Cannot read properties of undefined (reading '__computed_id')

test fails, I can remove my commented code but I wanted to show you both errors
1 creating a model in db instead of creating fake fields
2 tour test failure

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js (hunk 3 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: @chienandalu  yes, thanks to you too much for being interested.
Happy New Year

**Comment by CarlosRoca13** [comment]
I think that the real problem is that we don't set the method tearDownClass on test :sweat: 

Please add this code and try to run the test
```
    @classmethod
    def tearDownClass(cls):
        cls.loader.restore_registry()
        return super().tearDownClass()

**Comment by kobros-tech** [comment]
> I think that the real problem is that we don't set the method tearDownClass on test 😓
> 
> Please add this code and try to run the test
> 
> ```
>     @classmethod
>     def tearDownClass(cls):
>         cls.loader.restore_registry()
>         return super().tearDownClass()
> ```

ok, I will check on that 
thanks for helping us :)

**Comment by kobros-tech** [comment]
> @classmethod
>     def tearDownClass(cls):
>         cls.loader.restore_registry()
>         return super().tearDownClass()

@CarlosRoca13 

after I tried the code you gave to me I got this error:

2025-01-08 14:16:27,483 107694 INFO web_pivot_computed_measure_2 odoo.addons.web_pivot_computed_measure.tests.test_ui_pivot: ====================================================================== 
2025-01-08 14:16:27,483 107694 ERROR web_pivot_computed_measure_2 odoo.addons.web_pivot_computed_measure.tests.test_ui_pivot: FAIL: TestUIPivot.test_ui
Traceback (most recent call last):
  File "/home/kobros/Workspace/odoo18/web/web_pivot_computed_measure/tests/test_ui_pivot.py", line 68, in test_ui
    self.start_tour(
  File "/home/kobros/Workspace/odoo18/odoo/odoo/tests/common.py", line 2076, in start_tour
    return self.browser_js(url_path=url_path, code=code, ready=ready, timeout=timeout, success_signal="tour succeeded", **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kobros/Workspace/odoo18/odoo/odoo/tests/common.py", line 2052, in browser_js
    self.fail('%s\n\n%s' % (message, error))
AssertionError: The test code "odoo.startTour('web_pivot_computed_measure_tour', {"stepDelay": 100, "keepWatchBrowser": false, "debug": false, "startUrl": "/odoo", "delayToCheckUndeterminisms": 0})" failed

UncaughtTypeError: Cannot read properties of undefined (reading '__computed_id')
    at PivotModel.addComputedMeasure (http://127.0.0.1:8069/web/assets/ed10f81/web.assets_backend_lazy.min.js:375:229)
    at DropdownItemCustomMeasure.addMeasure (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:14672:150)
    at Object.mainEventHandler (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:1603:77)
    at HTMLButtonElement.listener (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:762:15)
    at dispatch (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15204:71)
    at triggerClick (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15132:94)
    at _pointerUp (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15187:24)
    at async _click (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15141:456)
    at async Object.click (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15197:187)
    at async TourHelpers.click (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:14733:72)
	at addComputedMeasure (http://127.0.0.1:8069/web/assets/ed10f81/web.assets_backend_lazy.min.js:374:228)
	at addMeasure (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:14671:149)
	at mainEventHandler (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:1602:76)
	at listener (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:761:14)
	at dispatch (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15203:70)
	at triggerClick (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15131:93)
	at _pointerUp (http://127.0.0.1:8069/web/assets/c4c2660/web.assets_web.min.js:15186:23)


Does it mean that test is not working well as the UI is not clicked and the field value is not assigned, shall we change the test or remote it better?

**Comment by kobros-tech** [comment]
@chienandalu 
@CarlosRoca13 

please review this PR and be aware that I made this PR to improve tests for this module that were failing in migration of web_responsive to 18.0

As you see now, the migration PR is depending on this PR and tests succeeded, thanks for your interset

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 6]
```diff
@@ -275,17 +279,27 @@ patch(PivotModel.prototype, {
...
+                    }
+                } else if (this.env.debug === "1") {
+                    // Console.warn(
+                    //     `Field "${fieldName}" is missing __computed_id`,
+                    //     field
+                    // );
                 }
+            } else if (this.env.debug === "1") {
+                // Console.warn(`Field "${fieldName}" is undefined in metaData.fields`);
             }
```
Comment (CarlosRoca13) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: I think this shoud be removed

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 7]
```diff
@@ -275,17 +279,27 @@ patch(PivotModel.prototype, {
...
-                const cm = this._computed_measures.find((item) => {
-                    return item.id === field.__computed_id;
-                });
-                if (!cm) {
-                    delete this.metaData.fields[fieldName];
-                    delete this.metaData.measures[fieldName];
-                    this.metaData.activeMeasures = this.metaData.activeMeasures.filter(
-                        (item) => item !== fieldName
+            if (field) {
+                if (field?.__computed_id) {
```
Comment (CarlosRoca13) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: You are verifying the field twice

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 8]
```diff
@@ -275,17 +279,27 @@ patch(PivotModel.prototype, {
...
+                            this.metaData.activeMeasures.filter(
+                                (item) => item !== fieldName
+                            );
+                    }
+                } else if (this.env.debug === "1") {
+                    // Console.warn(
+                    //     `Field "${fieldName}" is missing __computed_id`,
+                    //     field
+                    // );
                 }
```
Comment (CarlosRoca13) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: I think this shoud be removed

**Review by CarlosRoca13 — CHANGES_REQUESTED** [review]

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js (hunk 7 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: done

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js (hunk 6 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: done

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js (hunk 8 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]: done

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 9]
```diff
@@ -275,17 +279,20 @@ patch(PivotModel.prototype, {
...
                 if (!cm) {
                     delete this.metaData.fields[fieldName];
                     delete this.metaData.measures[fieldName];
                     this.metaData.activeMeasures = this.metaData.activeMeasures.filter(
                         (item) => item !== fieldName
                     );
                 }
+            } else if (this.env.debug === "1") {
+                // You can console log for debugging metaData.fields
+                // or fieldName or __computed_id
```
Comment (CarlosRoca13) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]:
```suggestion
            }
```
Remove unnecesary code

File: web_pivot_computed_measure/static/src/test/test.esm.js [hunk 10]
```diff
@@ -1,29 +1,18 @@
-/* Copyright 2022 Tecnativa - Carlos Roca
+/* Copyright 2022 Tecnativa - Carlos Roca, Kencove - Mohamed Alkobrosli
```
Comment (CarlosRoca13) [inline web_pivot_computed_measure/static/src/test/test.esm.js]:
```suggestion
/* Copyright 2022 Tecnativa - Carlos Roca
     Copyright 2025 Kencove - Mohamed Alkobrosli
```
Set the copyright on a different line with correct year

File: web_pivot_computed_measure/__manifest__.py [hunk 11]
```diff
@@ -7,10 +7,13 @@
     "author": "Tecnativa, Odoo Community Association (OCA)",
     "license": "AGPL-3",
     "website": "https://github.com/OCA/web",
-    "depends": ["web"],
+    "depends": ["web", "web_tour"],
```
Comment (CarlosRoca13) [inline web_pivot_computed_measure/__manifest__.py]: `web_tour` dependency is needed?

File: web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js [hunk 12]
```diff
@@ -230,6 +232,8 @@ patch(PivotModel.prototype, {
                     }
                     toEnableFields.push(afield);
                     toEnable.push(toEnableFields);
+                } else if (this.env.debug === "1") {
+                    // You can console log for debugging fieldDef or metaData.fields
```
Comment (CarlosRoca13) [inline web_pivot_computed_measure/static/src/pivot/pivot_model.esm.js]:
```suggestion
                } 
```
Remove unnecesary code

**Review by CarlosRoca13 — CHANGES_REQUESTED** [review]

File: web_pivot_computed_measure/__manifest__.py (hunk 11 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/__manifest__.py]: how can this test run without "web_tour"?:

`registry.category("web_tour.tours").add("web_pivot_computed_measure_tour", {`

Comment (CarlosRoca13) [reply web_pivot_computed_measure/__manifest__.py]: Ok, It was working before because web_tour is auto_install. The change has sense

**Review by CarlosRoca13 — APPROVED** [review]
Thanks :smile:

File: web_pivot_computed_measure/__manifest__.py (hunk 11 above)
Comment (CarlosRoca13) [reply web_pivot_computed_measure/__manifest__.py]: So you can remove `web` dependency. Because it is redundant

File: web_pivot_computed_measure/README.rst [hunk 13]
```diff
@@ -109,6 +109,10 @@ Contributors
   - Ernesto Tejeda
   - Carlos Roca
 
+- `Kencove <https://www.kencove.com/>`__:
```
Comment (pedrobaeza) [inline web_pivot_computed_measure/README.rst]: I don't think this test fixing should modify contributors.

File: web_pivot_computed_measure/static/src/test/test.esm.js [hunk 14]
```diff
@@ -1,29 +1,21 @@
...
  * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html) */
 
+/* Copyright 2025 Kencove - Mohamed Alkobrosli
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html) */
+
 import {registry} from "@web/core/registry";
 
 registry.category("web_tour.tours").add("web_pivot_computed_measure_tour", {
-    url: "/odoo",
+    url: "/odoo?debug=1",
```
Comment (pedrobaeza) [inline web_pivot_computed_measure/static/src/test/test.esm.js]: Things shouldn't be done in debug mode.

File: web_pivot_computed_measure/tests/res_users_fake.py [hunk 15]
```diff
@@ -1,12 +0,0 @@
-# Copyright 2022 Tecnativa - Carlos Roca
```
Comment (pedrobaeza) [inline web_pivot_computed_measure/tests/res_users_fake.py:1]: Why removing this?

Comment (kobros-tech) [reply web_pivot_computed_measure/tests/res_users_fake.py:1]: the entire file is removed, as I improved the test cases.

This module is not a must to be crated for only test cases, even more it was the main cause that migration for other module not to be merged in 18.0

Comment (kobros-tech) [reply web_pivot_computed_measure/tests/res_users_fake.py:1]: I replaced this module with an xml file to use web tour on res.partner model instead of a creating new model unnecessarily.

File: web_pivot_computed_measure/static/src/test/test.esm.js (hunk 14 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/test/test.esm.js]: there is a test case in odoo repo using the debug same as what I did here, also debug is showing us errors that are not shown in normal mode.

now whether we use debug or not tests are fixed and no side effect will happen.

File: web_pivot_computed_measure/README.rst (hunk 13 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/README.rst]: nevermind, I am making this PR as the test of this module caused a critical issue in the entire project for 17.0 branch

I will remove the extra contributor.

Comment (kobros-tech) [reply web_pivot_computed_measure/README.rst]: done

File: web_pivot_computed_measure/static/src/test/test.esm.js (hunk 14 above)
Comment (kobros-tech) [reply web_pivot_computed_measure/static/src/test/test.esm.js]: done

**Comment by dnplkndll** [comment]
@kobros-tech anything else to review on here? I guess is blocking the 
- https://github.com/OCA/web/pull/3037

**Comment by kobros-tech** [comment]
> @kobros-tech anything else to review on here? I guess is blocking the
> 
> * [[18.0][MIG] web_responsive: Migration to 18.0 #3037](https://github.com/OCA/web/pull/3037)

for this PR no and also #3037 they should be merged.

**Comment by pedrobaeza** [comment]
I still don't see the need of this change, as #3057 is showing that you can get a green CI with no change, and this is changing the module dependencies.

You are also adding your copyright in existing files, where if any, they should contain the copyright of all the contributors.

**Comment by pedrobaeza** [comment]
I understand you are answering me in https://github.com/OCA/web/pull/3057#issuecomment-2758987678.

Why are you adding here `web_tour` as dependency?

**Comment by kobros-tech** [comment]
> I understand you are answering me in [#3057 (comment)](https://github.com/OCA/web/pull/3057#issuecomment-2758987678).
> 
> Why are you adding here `web_tour` as dependency?

It is already the dependency, I think they forgot to be added or can be installed be default in previous versions.

Let me to try one final time to run tests on a clean database without web_tour, if there is an issue I will post now.

**Comment by pedrobaeza** [comment]
OK, and please correct the copyright thing.

**Comment by kobros-tech** [comment]
> OK, and please correct the copyright thing.

You are correct, I added tour as I am used to test with --test-tags which is focusing and compact, I guess here we use --test-enable which is full complete

test tour will be installed by default so I will remove now, for the copyright I didn't catch up?

**Comment by kobros-tech** [comment]
> > OK, and please correct the copyright thing.
> 
> You are correct, I added tour as I am used to test with --test-tags which is focusing and compact, I guess here we use --test-enable which is full complete
> 
> test tour will be installed by default so I will remove now, for the copyright I didn't catch up?

I saw it ok, I will remove our name too

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge patch

**Comment by kobros-tech** [comment]
@pedrobaeza 

I think we are ready now
